birdset/datamodule/components/augmentations.py

```diff
@@ -861,22 +861,22 @@ def random_background(self, audio: Audio, target_num_samples: int) -> torch.Tens
             background_path = random.choice(self.background_paths)
             background_num_samples = audio.get_num_samples(background_path)
 
-            if background_num_samples > missing_num_samples:
+            # If the background sample is longer than what we need, extract the exact amount
+            if background_num_samples >= missing_num_samples:
                 sample_offset = random.randint(
                     0, background_num_samples - missing_num_samples
                 )
-                num_samples = missing_num_samples
                 background_samples = audio(
                     background_path,
                     sample_offset=sample_offset,
-                    num_samples=num_samples,
+                    num_samples=missing_num_samples,
                 )
-                missing_num_samples = 0
-            else:
-                background_samples = audio(background_path)
-                missing_num_samples -= background_num_samples
+                pieces.append(background_samples)
+                break
 
+            background_samples = audio(background_path)
             pieces.append(background_samples)
+            missing_num_samples -= background_samples.shape[-1]
```

Comment:
@XgamerTV Lets use the `background_num_samples` variable again, this looks IMHO more readable.
```suggestion
            missing_num_samples -= background_num_samples
```

Reply:
Agreed